### PR TITLE
pcapplusplus: add version v22.11 (old)

### PR DESCRIPTION
**Pull request by seladb**
Specify library name and version:  ** pcapplusplus/22.11**

- Update PcapPlusPlus to v22.11
- v22.11 moved to C++11 so the `pthreads` dependency is not needed anymore. Hence, the `all` folder was renamed to `22.05` (which applies to all versions <= 22.05), and a new `22.11` folder was added which includes an updated `conanfile.py` which removes the `pthreads` dependency, and removed unnecessary patches
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/pcapplusplus/22.11/conanfile.py
Comment (samuel-emrys) [inline recipes/pcapplusplus/22.11/conanfile.py:61]: This goes into the `generate()` method in conan 2.0. Also, I think it would be my preference to explicitly enumerate the configure args rather than deferring to a script because it makes it clearer what the recipe is actually doing, but I'll let a maintainer chime in on whether this is acceptable practice.

File: recipes/pcapplusplus/22.11/conanfile.py
Comment (samuel-emrys) [inline recipes/pcapplusplus/22.11/conanfile.py:31]: I don't think this is necessary if you're using `Autotools`?

**Review by samuel-emrys — CHANGES_REQUESTED** [review]
Started to comment on some of the changes required to be compatible for the upcoming conan 2.0, and realised that a lot of this will need to be migrated - take a look at the [Conan 2.0 migration guide](https://docs.conan.io/en/latest/conan_v2.html) to see what changes will be required

**Comment by seladb** [comment]
> Started to comment on some of the changes required to be compatible for the upcoming conan 2.0, and realised that a lot of this will need to be migrated - take a look at the [Conan 2.0 migration guide](https://docs.conan.io/en/latest/conan_v2.html) to see what changes will be required

@samuel-emrys is it possible to merge this PR without Conan 2.0 migration and open a separate PR for the migration? I'm not familiar with Conan 2.0 and it'll take me a while to learn it. Also I think it's a better separate of concerns to do the pcapplusplus version bump and the migration in different PRs. Please let me know what you think

**Comment by samuel-emrys** [comment]
> > Started to comment on some of the changes required to be compatible for the upcoming conan 2.0, and realised that a lot of this will need to be migrated - take a look at the [Conan 2.0 migration guide](https://docs.conan.io/en/latest/conan_v2.html) to see what changes will be required
> 
> @samuel-emrys is it possible to merge this PR without Conan 2.0 migration and open a separate PR for the migration? I'm not familiar with Conan 2.0 and it'll take me a while to learn it. Also I think it's a better separate of concerns to do the pcapplusplus version bump and the migration in different PRs. Please let me know what you think

I'm not a maintainer, so I don't have the power to decide what gets approved in and what doesn't. I was just trying to give some rapid feedback given that the maintenance team are pretty swamped with the number of PRs at the moment. From what I've seen (and my experience), PRs aren't currently being approved without a 2.0 migration, but for an authoritative answer to your question you'll need to wait for this to get the attention of a maintainer. Honestly, it might just be faster to do the 2.0 migration. The migration guide that I linked is pretty thorough, and it doesn't take a long time to go through so I don't think it would be too much effort.

**Comment by seladb** [comment]
> > > Started to comment on some of the changes required to be compatible for the upcoming conan 2.0, and realised that a lot of this will need to be migrated - take a look at the [Conan 2.0 migration guide](https://docs.conan.io/en/latest/conan_v2.html) to see what changes will be required
> 
> > 
> 
> > @samuel-emrys is it possible to merge this PR without Conan 2.0 migration and open a separate PR for the migration? I'm not familiar with Conan 2.0 and it'll take me a while to learn it. Also I think it's a better separate of concerns to do the pcapplusplus version bump and the migration in different PRs. Please let me know what you think
> 
> 
> 
> I'm not a maintainer, so I don't have the power to decide what gets approved in and what doesn't. I was just trying to give some rapid feedback given that the maintenance team are pretty swamped with the number of PRs at the moment. From what I've seen (and my experience), PRs aren't currently being approved without a 2.0 migration, but for an authoritative answer to your question you'll need to wait for this to get the attention of a maintainer. Honestly, it might just be faster to do the 2.0 migration. The migration guide that I linked is pretty thorough, and it doesn't take a long time to go through so I don't think it would be too much effort.

Thanks for your response! Let's wait for a response from a maintainer, in the meantime if I have time I'll give a try to the Conan 2.0 migration

File: recipes/pcapplusplus/22.11/conanfile.py
Comment (prince-chrismc) [inline recipes/pcapplusplus/22.11/conanfile.py:1]: Is there a particular reason you added a new version recipe? General we try to have only 1 unless there's a huge difference...

I I paste https://raw.githubusercontent.com/conan-io/conan-center-index/96cfead04e245ba4243da176b07a1e4a879f6cef/recipes/pcapplusplus/22.05/conanfile.py over this file the diff only shows me 
image


Did you consider only using one recipe?

**Review by prince-chrismc — COMMENTED** [review]
Just one question to start

File: recipes/pcapplusplus/22.11/conanfile.py
Comment (seladb) [reply recipes/pcapplusplus/22.11/conanfile.py:1]: Thanks @prince-chrismc for your response!

As you can see from the diff file, the new version doesn't need `pthreads` as a dependency and it also needs fewer patches. I though it makes more sens to add a recipe than adding a lot of `if` statements in the current recipe. Please let me know what you think

Comment (prince-chrismc) [reply recipes/pcapplusplus/22.11/conanfile.py:1]: These are super close I think it can be just 1 recipe. It's just a few lines versus an entire recipe

The patches are listed in the conandata per version so that should be easy too

Comment (prince-chrismc) [reply recipes/pcapplusplus/22.11/conanfile.py:1]: I'd greatly appreciate if you could combine the two versions

Comment (seladb) [reply recipes/pcapplusplus/22.11/conanfile.py:1]: ok sure, I'll fix it 👍

Comment (seladb) [reply recipes/pcapplusplus/22.11/conanfile.py:1]: @prince-chrismc I opened a new PR with the fix: https://github.com/conan-io/conan-center-index/pull/14712

Can you review it?

**Pull request closed without merging**
